src/main.rs

```diff
@@ -315,13 +335,97 @@ impl<T> MaybeExternal<T> {
     }
 }
 
+fn infer_language(module: &wasmbin::Module) -> Result<Language> {
+    let mut imports = vec![];
+    let mut exports = vec![];
+
+    for section in &module.sections {
+        match section {
+            Section::Import(section) => {
+                let section = section.try_contents()?;
+                for import in section {
+                    imports.push(&import.path);
+                }
+            }
+            Section::Export(section) => {
+                let section = section.try_contents()?;
+                for export in section {
+                    exports.push(export);
+                }
+            }
+            _ => {}
+        }
+    }
+
+    // NOTE: Need to check for Blazor ahead of Emscripten
+    if imports.iter().any(|i| i.name.contains("blazor")) {
+        return Ok(Language::Blazor);
+    }
+
+    if imports.iter().any(|i| i.name.contains("emscripten")) {
+        return Ok(Language::Emscripten);
+    }
+
+    if imports.iter().any(|i| i.module == "go") {
+        return Ok(Language::Go);
+    }
+
+    // these are all based on Rust using wasm-bindgen
+    if imports.iter().any(|i| {
+        i.name.contains("wbindgen")
+            || i.name.contains("wbg")
+            || i.module == "wbg"
+            || i.module == "wbindgen"
+    }) || exports.iter().any(|e| e.name.contains("wbindgen"))
+    {
+        return Ok(Language::Rust);
+    }
+
+    // OK, so this one is *very* hacky! The hyphenate lib (https://github.com/mnater/Hyphenopoly) is found on a number of
+    // websites. It is written in AssemblyScript, and has a variety of different bundles. They all export the function
+    // 'hyphenate'.
+    // There is no other clear way of detecting AssemblyScript :-(
+    if exports.iter().any(|e| e.name == "hyphenate") {
```

Comment:
This one is too risky IMO. `hyphenate` might very well come from another language; I'd rather look at the whole URL like in the previous analysis to check if it's specifically the hyphenopoly library.